src/spath_mpi.c

```diff
@@ -23,11 +23,15 @@ Functions to send/recv paths with MPI
  * receivers must pass in a newly allocated path from spath_new() */
 int spath_bcast(spath* path, int root, MPI_Comm comm)
 {
-  /* if pointer is NULL, throw an error */
+  int need_to_abort = 0;
+  /* if pointer is NULL, throw an error and flag a need to abort*/
   if (path == NULL) {
-    fprintf(stderr, "NULL pointer passed for path @ %s:%d",
+    fprintf(stderr, "NULL pointer passed for path @ %s:%d\n",
       __FILE__, __LINE__
     );
+    return SPATH_FAILURE; 
```

Comment:
Similar MPI mismatch here we should watch out for.  We want to make sure all ranks behave consistently.

Reply:
For testing purposes, I'm thinking we may want to define something like an ``spath_assert(condition, comm, msg)``.  In production use, we could have this function print and error message and call MPI_Abort if the condition fails.  In testing mode, we could have this execute an allreduce to check the condition on *all* ranks, and if it fails on any rank, print an error, and then return a failure code.  We'd want something like this on all MPI-enabled components.